cmd/olclient/allegation.go

```diff
@@ -0,0 +1,150 @@
+		MaliciousAddress: args.MaliciousAddress,
+	}
+}
+
+var allegationCmd = &cobra.Command{
+	Use:   "allegation",
+	Short: "Perform allegation to a validator",
+	RunE:  allegationExec,
+}
+
+var allegationArgs = &AllegationArguments{}
+
+func init() {
+	EvidencesCmd.AddCommand(allegationCmd)
+	allegationCmd.Flags().BytesHexVar(&allegationArgs.Address, "address", []byte{}, "address for validator")
+	allegationCmd.Flags().StringVar(&allegationArgs.Password, "password", "", "password to access secure wallet")
+	allegationCmd.Flags().BytesHexVar(&allegationArgs.MaliciousAddress, "maliciousAddress", []byte{}, "malicious validator address")
+	allegationCmd.Flags().Int64Var(&allegationArgs.BlockHeight, "blockHeight", int64(0), "blockHeight for allegation")
+	allegationCmd.Flags().StringVar(&allegationArgs.ProofMsg, "proofMsg", "", "proofMsg for allegation")
+}
+
+func allegationExec(cmd *cobra.Command, args []string) error {
+	ctx := NewContext()
+	ctx.logger.Debug("Have Allegation Request", "allegationArgs", allegationArgs)
+
+	rootPath, err := filepath.Abs(rootArgs.rootDir)
+	if err != nil {
+		return err
+	}
+
+	cfg := &config.Server{}
+
+	//Prompt for password
+	if len(allegationArgs.Password) == 0 {
+		allegationArgs.Password = PromptForPassword()
+	}
+
+	//Create new Wallet and User Address
+	wallet, err := accounts2.NewWalletKeyStore(keyStorePath)
+	if err != nil {
+		ctx.logger.Error("failed to create secure wallet", err)
+		return err
+	}
+
+	//Verify User Password
+	usrAddress := keys.Address(allegationArgs.Address)
+	authenticated, err := wallet.VerifyPassphrase(usrAddress, allegationArgs.Password)
+	if !authenticated {
+		ctx.logger.Error("authentication error", err)
+		return err
+	}
+
+	err = cfg.ReadFile(cfgPath(rootPath))
+	if err != nil {
+		return errors.Wrapf(err, "failed to read configuration file at at %s", cfgPath(rootPath))
+	}
+
+	// Create message
+	fullnode := ctx.clCtx.FullNodeClient()
+
+	out, err := fullnode.Allegation(allegationArgs.ClientRequest())
+	if err != nil {
+		ctx.logger.Error("Error in applying ", err.Error())
+		return err
+	}
+
+	rawTx := &action.RawTx{}
+	err = serialize.GetSerializer(serialize.NETWORK).Deserialize(out.RawTx, rawTx)
+	if err != nil {
+		ctx.logger.Error("failed to deserialize RawTx", err)
+		return err
+	}
+
+	if !wallet.Open(usrAddress, allegationArgs.Password) {
+		ctx.logger.Error("failed to open secure wallet")
+		return err
+	}
+
+	//Sign Raw "Send" Transaction Using Secure Wallet.
+	pub, signature, err := wallet.SignWithAddress(out.RawTx, usrAddress)
+	if err != nil {
+		ctx.logger.Error("error signing transaction", err)
+	}
+
+	signedTx := &action.SignedTx{
+		RawTx: *rawTx,
+		Signatures: []action.Signature{
+			{Signer: pub, Signed: signature},
+		},
+	}
+
+	packet, err := serialize.GetSerializer(serialize.NETWORK).Serialize(signedTx)
+	if err != nil {
+		ctx.logger.Error("failed to serialize signedTx", err)
+		return err
+	}
+
+	result, err := ctx.clCtx.BroadcastTxCommit(packet)
```

Comment:
Can you use "BroadcastTxSync" instead. We recently changed this to prevent timeouts in production.